firmware/Core/Src/rtos_tasks/rtos_tasks.c

```diff
@@ -163,3 +165,49 @@ void TASK_execute_telecommands(void *argument) {
 
 	} /* End Task's Main Loop */
 }
+
+void TASK_receive_gps_info(void *argument) {
+	TASK_HELP_start_of_task();
+
+	char latest_gps_response[UART_gps_buffer_len];
+	uint16_t latest_gps_response_len = 0;
+
+	while (1) {
+
+		osDelay(200);
+
+		// Checking if there is data in the GPS buffer
+		if(UART_gps_buffer_write_idx > 0) {
```

Comment:
Isn't this going to potentially try to read the GPS data as it's still being sent? Other subsystems use a timeout to determine when it's done sending.

Reply:
So like I delay the system and then check if like new data has come in? That is my first thought on this. 


Ill try move forward with what I said above until you say otherwise.